File: Generating/abstractFactory/python/WindowsStore.py
from interfaces import AppInterface, GameInterface, StoreFactory

class WindowsGame(GameInterface):
  __params = None
  PLATFORM = 'Windows'
  
  def __init__(self, name):
    self.__params = {
      'name': name,
      'type': 'game'
    }

    print('Game ', str(self.__params['name']), ' is created')    
    pass
  
  def play(self):
    print('Game ', str(self.__params['name']), ' is started!')
    pass
  
  def getInfo(self):
    return self.__params
  
  def printInformation(self):
    print('PLATFORM: ', str(WindowsGame.PLATFORM))
    print(
      'GAME NAME: ', str(self.__params['name']), '\n'
      'TYPE: ', str(self.__params['type']), '\n'
    )
    pass

class WindowsApp(AppInterface):
  __params = None
  PLATFORM = 'Windows'
  DEFAULTPRICE = 10

  def __init__(self, name):
    self.__params = {
      'name': name,
      'type': 'app',
      'price': WindowsApp.DEFAULTPRICE
    }
    
    print('App ', str(self.__params['name']), ' is created')
    print(
      'APP NAME: ', str(self.__params['name']), '\n'
      'TYPE: ', str(self.__params['type']), '\n'
    )
    pass

  def getInfo(self):
    return self.__params
  
  def start(self):
    print('App ', str(self.__params['name']), ' is started')
    pass
  
  def buyApp(self, deposit):
    if deposit < self.__params['price']:
      raise ValueError(
        'your deposit less your: ', str(deposit),
        ' needed: ', str(self.__params['price'])
      )

    print(
      'App ', str(self.__params['name']),
      ' purchased for ', str(WindowsApp.PLATFORM),
      ' written off ', str(self.__params['price'])
    )
    
    return True

  def printInformation(self):
    print('PLATFORM: ', str(WindowsApp.PLATFORM))
    print(
      'APP NAME: ', str(self.__params['name']), '\n'
      'TYPE: ', str(self.__params['type']), '\n'
      'APP PRICE: ', str(self.__params['price']), '\n'
    )
    pass

class WindowsStore(StoreFactory):
  def createGame(self, name):
    return WindowsGame(name)
  def createApplication(self, name):
    return WindowsApp(name)

File: Generating/abstractFactory/cpp/windows_store.cpp
#ifndef windows_store
#define windows_store

#include <tuple>
#include <string>
#include <iostream>
#include <stdexcept>
#include "interfaces.hpp"

using namespace std;

class WindowsGame : public GameInterface
{
  private:
    std::string name;
    std::string type = "game";
  public:
    WindowsGame(std::string name) {
      this->name = name;
    }

    std::tuple<std::string, std::string> getInfo() {
      return std::make_tuple(this->name, this->type);
    }

    void play() {
      cout << "Game " << this->name << " runing" << endl;
    }

    void printInformation() {
      cout << "PLATFORM: Windows" << endl;
      cout << "NAME:     " << this->name << endl;
      cout << "TYPE:     " << this->type << endl;
    }
};

class WindowsApp : public AppInterface
{
  private:
    std::string name;
    std::string type = "app";
    int default_price = 10;
  public:
    WindowsApp(std::string name) {
      this->name = name;
    }

    std::tuple<std::string, std::string, int> getInfo() {
      return std::make_tuple(this->name, this->type, this->default_price);
    }

    bool buyApp(int deposit) {
      if (this->default_price != deposit) {
        throw std::invalid_argument("deposit less");
      }

      cout << "App " << this->name << " buyed success" << endl;
      return true;
    }

    void start() {
      cout << "App " << this->name << " started" << endl;
    }

    void printInformation() {
      cout << "PLATFORM: Windows" << endl;
      cout << "NAME:     " << this->name << endl;
      cout << "TYPE:     " << this->type << endl;
      cout << "PRICE:    " << this->default_price << endl;
    }
};

class WindowsStore : public StoreFactory
{
  public:
    AppInterface * createApp(std::string name) {
      return new WindowsApp(name);
    }

    GameInterface * createGame(std::string name) {
      return new WindowsGame(name);
    }
};

#endif